mixer/test/spybackend/nosession_integration_test.go

```diff
@@ -54,14 +214,155 @@ func TestNoSessionBackend(t *testing.T) {
 				_ = ctx.(Server).Close()
 			},
 			GetState: func(ctx interface{}) (interface{}, error) {
+				// TODO return the state of data received by the spy adapter; deterministic order.
 				return nil, validateNoSessionBackend(ctx, t)
 			},
-			ParallelCalls: []adapter_integration.Call{},
-			Configs:       []string{},
-			Want: `{
-              "AdapterState": null,
-		      "Returns": []
-            }`,
+			ParallelCalls: []adapter_integration.Call{
+				// 3 report calls; varying request.size attribute and no attributes call too.
+				{
+					CallKind: adapter_integration.REPORT,
+					Attrs:    map[string]interface{}{"request.size": 666},
+				},
+				{
+					CallKind: adapter_integration.REPORT,
+					Attrs:    map[string]interface{}{"request.size": 888},
+				},
+				{
+					CallKind: adapter_integration.REPORT,
+				},
+
+				// 3 check calls; varying source.name attribute and no attributes call too.,
+				{
+					CallKind: adapter_integration.CHECK,
+					Attrs:    map[string]interface{}{"source.name": "foobar"},
+				},
+				{
+					CallKind: adapter_integration.CHECK,
+					Attrs:    map[string]interface{}{"source.name": "bazbaz"},
+				},
+				{
+					CallKind: adapter_integration.CHECK,
+				},
+
+				// one call with quota args
+				{
+					CallKind: adapter_integration.CHECK,
+					Quotas: map[string]istio_mixer_v1.CheckRequest_QuotaParams{
+						"requestCount": {
+							Amount:     30,
+							BestEffort: true,
+						},
+					},
+				},
+
+				// one report request with request.id to match r4 rule
+				{
+					CallKind: adapter_integration.REPORT,
+					Attrs:    map[string]interface{}{"request.id": "somereqid"},
+				},
+			},
+			GetConfig: func(ctx interface{}) ([]string, error) {
```

Comment:
the static config was not right ? Is there a reason to make it dynamic via GetConfig ?

Reply:
Handler config has connection config which is only available after the server starts.

Reply:
I think we should get rid of static `Configs` completely then for integration tests. I will do that. All tests can just pass via your GetConfig call back.